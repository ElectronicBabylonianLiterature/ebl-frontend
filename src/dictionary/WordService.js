class WordService {
  constructor(wordRepository) {
    this.wordRepository = wordRepository
  }

  find(id) {
    return this.wordRepository.find(id)
  }

  search(query) {
    return this.wordRepository.search(query)
  }

  update(word) {
    return this.wordRepository.update(word)
  }
}

export default WordService
