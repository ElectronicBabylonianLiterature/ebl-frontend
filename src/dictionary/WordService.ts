class WordService {
  private readonly wordRepository

  constructor(wordRepository) {
    this.wordRepository = wordRepository
  }

  find(id) {
    return this.wordRepository.find(id)
  }

  search(query) {
    return this.wordRepository.search(query)
  }

  update(word) {
    return this.wordRepository.update(word)
  }
}

export default WordService
