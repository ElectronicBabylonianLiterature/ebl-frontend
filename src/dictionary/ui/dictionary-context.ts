import WordService from 'dictionary/application/WordService'
import React, { useContext } from 'react'

export const DictionaryContext = React.createContext<WordService | null>(null)

export function useDictionary(): WordService {
  const dictionary = useContext(DictionaryContext)
  if (dictionary === null) {
    throw new Error('useDictionary must be inside DictionaryContext.Provider')
  } else {
    return dictionary
  }
}
