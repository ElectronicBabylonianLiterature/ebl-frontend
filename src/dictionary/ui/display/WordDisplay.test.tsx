import React, { ReactNode } from 'react'
import { render, screen } from '@testing-library/react'
import { MemoryRouter, Route, RouteComponentProps } from 'react-router-dom'
import SessionContext from 'auth/SessionContext'
import WordDisplay from 'dictionary/ui/display/WordDisplay'
import WordService from 'dictionary/application/WordService'
import TextService from 'corpus/application/TextService'
import MemorySession from 'auth/Session'
import Bluebird from 'bluebird'
import { dictionaryLineDisplayFactory } from 'test-support/chapter-fixtures'
import { DictionaryContext } from '../dictionary-context'
import { Chance } from 'chance'

jest.mock('dictionary/application/WordService')
const wordService = new (WordService as jest.Mock<jest.Mocked<WordService>>)()

jest.mock('corpus/application/TextService')
const textService = new (TextService as jest.Mock<jest.Mocked<TextService>>)()

const session = new MemorySession(['read:words'])

const chance = new Chance('word-display-test')

const word = {
  lemma: ['oheto', 'ofobuv'],
  derived: [
    [
      {
        lemma: ['pabemiiwu', 'pa'],
        notes: ['tin', 'mi'],
        homonym: 'II',
      },
      {
        lemma: ['tesbu', 'pi'],
        notes: ['cujuwibi', 'vu'],
        homonym: 'IV',
      },
    ],
    [
      {
        lemma: ['utgibhev', 'ra'],
        notes: ['sulhe', 'ulkil'],
        homonym: 'II',
      },
      {
        lemma: ['tokteuf', 'nowu'],
        notes: ['ziwaini', 'reguwza'],
        homonym: 'I',
      },
    ],
  ],
  _id: 'c49ca1a96f9ce4d45bdc9962307a703eba53c0a8',
  attested: true,
  legacyLemma: 'ri',
  homonym: 'IV',
  meaning:
    'Bo hafe jed egeragpo viwidih zutu we ijcegwoj opuziw agse wad eksorep edzu tuum za.',
  pos: ['MOD', 'PRP'],
  source: '**source**',
  guideWord: 'fe',
  arabicGuideWord: 'fe',
  origin: 'cda',
  cdaAddenda: 'fe',
  supplementsAkkadianDictionaries: 'word',
  oraccWords: [
    {
      lemma: 'serbig',
      guideWord: 'mut',
    },
    {
      lemma: 'sujemges',
      guideWord: 'ak',
    },
  ],
  forms: [
    {
      lemma: ['gajremdac', 'la'],
      notes: ['pe', 'usrew'],
      attested: false,
    },
    {
      lemma: ['rap', 'oneun'],
      notes: ['gi', 'sobkuva'],
      attested: false,
    },
  ],
  logograms: [
    {
      notes: ['zuisi', 'lan'],
      logogram: ['echo', 'bravo'],
    },
    {
      notes: ['ofinijvef', 'lo'],
      logogram: ['bravo', 'charlie'],
    },
  ],
  derivedFrom: {
    lemma: ['jaruw', 'itiparu'],
    notes: ['muvi', 'panum'],
    homonym: 'III',
  },
  akkadischeGlossareUndIndices: [
    {
      mainWord: '<sup style="font-style: normal;">serbig</sup>',
      note: '<sup style="font-style: normal;">jaruw</sup>',
      reference: '<em>itiparu</em>',
      AfO: 'panum',
      agiID: '17865',
    },
    {
      mainWord: 'puul',
      note: 'aasad asas <em>itiparu</em>',
      reference: '<i>širkū</i>',
      AfO: 'juzz',
      agiID: '87161',
    },
  ],
  amplifiedMeanings: [
    {
      meaning:
        'La gumib not ukugub niruhi majiv aruha ro zar zi adiazku tiwom wezil.',
      key: 'Dtt',
      vowels: [
        {
          value: ['a', 'e'],
          notes: ['walzec', 'mawif'],
        },
        {
          value: ['u', 'a'],
          notes: ['novi', 'gi'],
        },
      ],
      entries: [
        {
          meaning:
            'Mu efobizro cafofiri mub kes dovlulle gow tojovdad kiogfuz mupasa viegsif rogo tep aku resbeju wop ubro luez.',
          vowels: [
            {
              value: ['a', 'e'],
              notes: ['jefizo', 'kosgunra'],
            },
            {
              value: ['a', 'u'],
              notes: ['tiji', 'fenuaf'],
            },
          ],
        },
        {
          meaning:
            'Lahwev memzubam nusoge ebcem nifta wojfobop mikgu nus fohom cifa safezot wuot va ebo widmoc.',
          vowels: [
            {
              value: ['a', 'u'],
              notes: ['avadin', 'ja'],
            },
            {
              value: ['i', 'e'],
              notes: ['bur', 'louwadeg'],
            },
          ],
        },
      ],
    },
    {
      meaning:
        'Cafheejo juud ulzi we udubu odejibviz uhciw ge citmi hi uphowge tiri tatwovef hi dowuile dih wacbus najab.',
      key: 'N',
      vowels: [
        {
          value: ['e', 'u'],
          notes: ['seppoz', 'josewe'],
        },
        {
          value: ['u', 'e'],
          notes: ['tejfesag', 'mo'],
        },
      ],
      entries: [
        {
          meaning:
            'Levnevse irizob tetah radjic nagcir biiv ragow tu wit wiahahi miw pa kobjawib hol ilobelhon of soh.',
          vowels: [
            {
              value: ['i', 'e'],
              notes: ['juwufan', 'de'],
            },
            {
              value: ['u', 'u'],
              notes: ['macaju', 'leudva'],
            },
          ],
        },
        {
          meaning:
            'Cuttegav alo oluusute era esudoj futasuj ugha ecosu bueciri was jahi mozoscu ditniv vorun.',
          vowels: [
            {
              value: ['u', 'e'],
              notes: ['furoli', 'tum'],
            },
            {
              value: ['i', 'u'],
              notes: ['fislagji', 'ju'],
            },
          ],
        },
      ],
    },
  ],
}

let container: HTMLElement

describe('Fetch word', () => {
  beforeEach(async () => {
    const genres = ['L', 'D', 'Lex', 'Med']
    wordService.find.mockReturnValue(Bluebird.resolve(word))
    textService.searchLemma.mockReturnValue(
      Bluebird.resolve([
        dictionaryLineDisplayFactory.build(
          {},
          { transient: { chance: chance } }
        ),
      ])
    )
    renderWordInformationDisplay()
    await screen.findByText(word.meaning)
    expect(wordService.find).toBeCalledWith('id')

    genres.forEach((genre) => {
      expect(textService.searchLemma).toBeCalledWith(word._id, genre)
    })
  })
  it('correctly displays word parts', async () => {
    await screen.findAllByText(new RegExp(word.guideWord))
    expect(container).toMatchSnapshot()
  })
})

function renderWordInformationDisplay() {
  container = render(
    <MemoryRouter initialEntries={['/dictionary/id']}>
      <SessionContext.Provider value={session}>
        <Route
          path="/dictionary/:id"
          render={(props: RouteComponentProps<{ id: string }>): ReactNode => (
            <DictionaryContext.Provider value={wordService}>
              <WordDisplay
                textService={textService}
                wordService={wordService}
                {...props}
              />
            </DictionaryContext.Provider>
          )}
        />
      </SessionContext.Provider>
    </MemoryRouter>
  ).container
}
