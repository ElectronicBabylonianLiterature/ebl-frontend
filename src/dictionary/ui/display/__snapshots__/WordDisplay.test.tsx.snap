// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch word correctly displays word parts 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="main__content"
    >
      <header
        class="main__header"
      >
        <nav
          aria-label="breadcrumb"
          class="main__breadcrumbs"
        >
          <ol
            class="breadcrumb"
          >
            <li
              class="breadcrumb-item active"
            >
              <a
                href="/"
              >
                eBL
              </a>
            </li>
            <li
              class="breadcrumb-item active"
            >
              <a
                href="/dictionary"
              >
                Dictionary
              </a>
            </li>
            <li
              aria-current="page"
              class="breadcrumb-item active"
            >
              c49ca1a96f9ce4d45bdc9962307a703eba53c0a8
            </li>
          </ol>
        </nav>
        <h2
          class="main__heading"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <strong>
                oheto ofobuv
                 
                IV
              </strong>
              , “
              <span>
                fe
              </span>
              ”
            </div>
            <div
              class="col"
            >
              <h5
                class="text-secondary"
              >
                (
                MOD, PRP
                )
              </h5>
            </div>
            <div
              class="pr-5 mr-5 col-auto"
            >
              <div
                class="border border-dark p-1 text-secondary h6"
              >
                +oheto[fe]MOD$
                <button
                  class="ml-2 copyIcon btn btn-primary"
                  type="button"
                >
                  <i
                    class="fas fa-copy"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="arabicGuideWord col"
            >
              fe
            </div>
          </div>
        </h2>
        <div
          class="main__toolbar btn-toolbar"
          role="toolbar"
        />
      </header>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h3
            class="dictionary-heading"
            id="Ⅰ"
          >
            Ⅰ
            . 
            A Concise Dictionary of Akkadian
          </h3>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="row"
          >
            <div
              class="col offset-1"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Other forms: 
                   
                  <span>
                    <span>
                      pe 
                    </span>
                    <span>
                      <em>
                        gajremdac** 
                      </em>
                      la**
                    </span>
                     
                     
                    <span>
                      usrew
                    </span>
                    
                  </span>
                  , 
                  <span>
                    <span>
                      gi 
                    </span>
                    <span>
                      <em>
                        rap** 
                      </em>
                      oneun**
                    </span>
                     
                     
                    <span>
                      sobkuva
                    </span>
                    
                  </span>
                  
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Attested stems: 
                  <span>
                    <a
                      href="/dictionary/c49ca1a96f9ce4d45bdc9962307a703eba53c0a8#attested-stem-0"
                    >
                      Dtt
                    </a>
                    , 
                  </span>
                  <span>
                    <a
                      href="/dictionary/c49ca1a96f9ce4d45bdc9962307a703eba53c0a8#attested-stem-1"
                    >
                      N
                    </a>
                  </span>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Derivatives: 
                   
                  <span>
                    <span>
                      tin 
                    </span>
                    <a
                      href="/dictionary/pabemiiwu%20pa%20II"
                    >
                      <em>
                        pabemiiwu pa
                      </em>
                    </a>
                     
                    II
                     
                     
                    <span>
                      mi
                    </span>
                    
                  </span>
                  , 
                  <span>
                    <span>
                      cujuwibi 
                    </span>
                    <a
                      href="/dictionary/tesbu%20pi%20IV"
                    >
                      <em>
                        tesbu pi
                      </em>
                    </a>
                     
                    IV
                     
                     
                    <span>
                      vu
                    </span>
                    
                  </span>
                  
                  ; 
                   
                  <span>
                    <span>
                      sulhe 
                    </span>
                    <a
                      href="/dictionary/utgibhev%20ra%20II"
                    >
                      <em>
                        utgibhev ra
                      </em>
                    </a>
                     
                    II
                     
                     
                    <span>
                      ulkil
                    </span>
                    
                  </span>
                  , 
                  <span>
                    <span>
                      ziwaini 
                    </span>
                    <a
                      href="/dictionary/tokteuf%20nowu%20I"
                    >
                      <em>
                        tokteuf nowu
                      </em>
                    </a>
                     
                    I
                     
                     
                    <span>
                      reguwza
                    </span>
                    
                  </span>
                  
                  ; 
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Derived from: 
                  <span>
                    <span>
                      muvi 
                    </span>
                    <a
                      href="/dictionary/jaruw%20itiparu%20III"
                    >
                      <em>
                        jaruw itiparu
                      </em>
                    </a>
                     
                    III
                     
                     
                    <span>
                      panum
                    </span>
                    
                  </span>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Logograms: 
                   
                  <span>
                    <span>
                      zuisi 
                    </span>
                    <span>
                      echo
                    </span>
                     
                    <span>
                       
                      <span>
                        bravo
                      </span>
                      
                       
                    </span>
                     
                     
                    <span>
                      lan
                    </span>
                    
                  </span>
                  , 
                  <span>
                    <span>
                      ofinijvef 
                    </span>
                    <span>
                      bravo
                    </span>
                     
                    <span>
                       
                      <span>
                        charlie
                      </span>
                      
                       
                    </span>
                     
                     
                    <span>
                      lo
                    </span>
                    
                  </span>
                  
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col"
                />
              </div>
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  <br />
                  <span>
                    Bo hafe jed egeragpo viwidih zutu we ijcegwoj opuziw agse wad eksorep edzu tuum za.
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
            >
              <div
                class="col"
              >
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <div
                      class="row"
                    >
                      <strong>
                        Dtt
                      </strong>
                         
                       
                      <p>
                        La gumib not ukugub niruhi majiv aruha ro zar zi adiazku tiwom wezil.
                      </p>
                    </div>
                    <div
                      class="row"
                    >
                      <ol>
                        <li
                          id="attested-stem-0"
                        >
                          <div
                            class="ml-3"
                          >
                            <span>
                              Mu efobizro cafofiri mub kes dovlulle gow tojovdad kiogfuz mupasa viegsif rogo tep aku resbeju wop ubro luez.
                            </span>
                          </div>
                        </li>
                        <li
                          id="attested-stem-1"
                        >
                          <div
                            class="ml-3"
                          >
                            <span>
                              Lahwev memzubam nusoge ebcem nifta wojfobop mikgu nus fohom cifa safezot wuot va ebo widmoc.
                            </span>
                          </div>
                        </li>
                      </ol>
                    </div>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col"
                  >
                    <div
                      class="row"
                    >
                      <strong>
                        N
                      </strong>
                         
                       
                      <p>
                        Cafheejo juud ulzi we udubu odejibviz uhciw ge citmi hi uphowge tiri tatwovef hi dowuile dih wacbus najab.
                      </p>
                    </div>
                    <div
                      class="row"
                    >
                      <ol>
                        <li
                          id="attested-stem-0"
                        >
                          <div
                            class="ml-3"
                          >
                            <span>
                              Levnevse irizob tetah radjic nagcir biiv ragow tu wit wiahahi miw pa kobjawib hol ilobelhon of soh.
                            </span>
                          </div>
                        </li>
                        <li
                          id="attested-stem-1"
                        >
                          <div
                            class="ml-3"
                          >
                            <span>
                              Cuttegav alo oluusute era esudoj futasuj ugha ecosu bueciri was jahi mozoscu ditniv vorun.
                            </span>
                          </div>
                        </li>
                      </ol>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
            />
          </div>
        </div>
      </div>
      <div
        class="text-center border border-dark mx-5 m-3 p-2"
      >
        <strong>
          From
        </strong>
        <br />
        Black, J.; George, A.R.; Postgate, N.
        , 
        <em>
          A Concise Dictionary of Akkadian
        </em>
        . 
        Second (corrected) printing. SANTAG Arbeiten und Untersuchungen zur Keilschriftkunde 5. Wiesbaden: Harrassowitz, ²2000
        .
        <br />
        <strong>
          By permission from Harrassowitz
        </strong>
        <br />
        <a
          class="text-dark "
          href="https://www.harrassowitz-verlag.de/isbn_978-3-447-04264-2.ahtml"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            class="pointer__hover my-2 fas fa-shopping-cart fa-2x"
          />
        </a>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h3
            class="dictionary-heading"
            id="Ⅱ"
          >
            Ⅱ
            . 
            A Concise Dictionary of Akkadian (Justifications, Addenda and Corrigenda)
          </h3>
        </div>
      </div>
      <div
        class="ml-5 row"
      >
        <div
          class="col"
        >
           
          <span>
            fe
          </span>
        </div>
      </div>
      <div
        class="text-center border border-dark mx-5 m-3 p-2"
      >
        <strong>
          From
        </strong>
        <br />
        Postgate, N.; Worthington, M.
        , 
        <em>
          A Concise Dictionary of Akkadian (Justifications, Addenda and Corrigenda)
        </em>
        . 
        2003–2011
        .
        <br />
        <strong>
          By permission from the authors
        </strong>
        <br />
        <a
          class="text-dark "
          href="https://www.soas.ac.uk/cda-archive/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            class="pointer__hover my-2 fas fa-external-link-square-alt"
          />
        </a>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h3
            class="dictionary-heading"
            id="Ⅲ"
          >
            Ⅲ
            . 
            Akkadische Glossare und Indizes
          </h3>
        </div>
      </div>
      <div
        class="ml-5 row"
      >
        <div
          class="col"
        >
          <div
            class="small text-black-50 row"
          >
            <div>
              <sup
                style="font-style: normal;"
              >
                serbig
              </sup>
            </div>
          </div>
          <div
            class="row"
          >
            <div>
              <sup
                style="font-style: normal;"
              >
                jaruw
              </sup>
            </div>
          </div>
          <div
            class="row"
          >
            <div>
              <em>
                itiparu
              </em>
            </div>
          </div>
          <div
            class="mb-3 row"
          >
            <div
              class="small text-black-50 ml-3"
            >
              [
              panum
              ]
            </div>
          </div>
        </div>
      </div>
      <div
        class="ml-5 row"
      >
        <div
          class="col"
        >
          <div
            class="small text-black-50 row"
          >
            <div>
              puul
            </div>
          </div>
          <div
            class="row"
          >
            <div>
              aasad asas 
              <em>
                itiparu
              </em>
            </div>
          </div>
          <div
            class="row"
          >
            <div>
              <i>
                širkū
              </i>
            </div>
          </div>
          <div
            class="mb-3 row"
          >
            <div
              class="small text-black-50 ml-3"
            >
              [
              juzz
              ]
            </div>
          </div>
        </div>
      </div>
      <div
        class="text-center border border-dark mx-5 m-3 p-2"
      >
        <strong>
          From
        </strong>
        <br />
        Sommerfeld, W.
        , 
        <em>
          Akkadische Glossare und Indizes
        </em>
        . 
        Version 1.1 (26. Mai 2021)
        .
        <br />
        <a
          class="text-dark "
          href="https://creativecommons.org/licenses/by-nd/4.0/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <strong>
            CC BY-ND 4.0
          </strong>
        </a>
        <br />
        <a
          class="text-dark "
          href="https://www.uni-marburg.de/cnms/forschung/dnms/apps/agi"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            class="pointer__hover my-2 fas fa-external-link-square-alt"
          />
        </a>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h3
            class="dictionary-heading"
            id="Ⅳ"
          >
            Ⅳ
            . 
            Supplement to the Akkadian Dictionaries
          </h3>
        </div>
      </div>
      <div
        class="supplementsAkkadianDictionaries row"
      >
        <div
          class="col"
        >
           
          <span>
            word
          </span>
        </div>
      </div>
      <div
        class="text-center border border-dark mx-5 m-3 p-2"
      >
        <strong>
          From
        </strong>
        <br />
        Streck, M.P.
        , 
        <em>
          Supplement to the Akkadian Dictionaries
        </em>
        . 
        2013–
        .
        <br />
        <a
          class="text-dark "
          href="https://creativecommons.org/licenses/by-sa/3.0/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <strong>
            CC BY-ND 3.0
          </strong>
        </a>
        <br />
        <a
          class="text-dark "
          href="https://altorient.gko.uni-leipzig.de/etymd.html"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            class="pointer__hover my-2 fas fa-external-link-square-alt"
          />
        </a>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <h3
            class="dictionary-heading"
            id="Ⅴ"
          >
            Ⅴ
            . 
            Corpus
          </h3>
        </div>
      </div>
      <nav
        class="nav nav-tabs"
        role="tablist"
      >
        <a
          aria-selected="true"
          class="nav-item nav-link active"
          data-rb-event-key="L"
          href="#"
          role="tab"
        >
          Literature
        </a>
        <a
          aria-selected="false"
          class="nav-item nav-link"
          data-rb-event-key="D"
          href="#"
          role="tab"
        >
          Divination
        </a>
        <a
          aria-selected="false"
          class="nav-item nav-link"
          data-rb-event-key="Lex"
          href="#"
          role="tab"
        >
          Lexicography
        </a>
        <a
          aria-selected="false"
          class="nav-item nav-link"
          data-rb-event-key="Med"
          href="#"
          role="tab"
        >
          Medicine
        </a>
      </nav>
      <div
        class="tab-content"
      >
        <div
          aria-hidden="false"
          class="fade tab-pane active show"
          role="tabpanel"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <table>
                <tbody>
                  <tr>
                    <th
                      class="lines-with-lemma__genre"
                      colspan="2"
                      scope="col"
                    >
                      Neo-Assyrian
                    </th>
                  </tr>
                  <tr>
                    <th
                      class="lines-with-lemma__header"
                      colspan="2"
                      scope="col"
                    >
                      <span
                        class="lines-with-lemma__textname"
                      >
                        Wama uhro etacafnez ovu wopano vasozli behe halecez pur honuv ziv upepufa rezami ficfiozi veh ti ta.
                      </span>
                       
                      <span>
                        (
                        <a
                          href="https://www.ebl.lmu.de/corpus/Lex/2771/1007298344386560/NA/Pahpa%20muwupdi%20bucelo%20du%20siwele%20mubupel%20januw%20hencan%20iwok%20zo%20fuizi%20pedo%20kemum%20jewkudin%20feksukmab%20bukrepcu."
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          Lex
                           
                          MMDCCLXXI.1007298344386560
                        </a>
                        )
                      </span>
                       
                      <span>
                        Pahpa muwupdi bucelo du siwele mubupel januw hencan iwok zo fuizi pedo kemum jewkudin feksukmab bukrepcu.
                      </span>
                    </th>
                  </tr>
                  <tr
                    class="lines-with-lemma__textline"
                  >
                    <td>
                      <a
                        href="https://www.ebl.lmu.de/corpus/Lex/2771/1007298344386560/NA/Pahpa%20muwupdi%20bucelo%20du%20siwele%20mubupel%20januw%20hencan%20iwok%20zo%20fuizi%20pedo%20kemum%20jewkudin%20feksukmab%20bukrepcu.#1"
                      >
                        1
                      </a>
                    </td>
                    <td>
                      <span
                        class="Transliteration__wordSeparator Transliteration__wordSeparator--AKKADIAN"
                      >
                         
                      </span>
                      <span
                        class="Transliteration__AkkadianWord Transliteration__AkkadianWord--AKKADIAN"
                      >
                        <span
                          class=""
                        >
                          <span
                            class="Transliteration__ValueToken"
                          >
                            <span
                              class=""
                            >
                              kur-kur
                            </span>
                          </span>
                          <sup
                            class="Transliteration__flag"
                          />
                        </span>
                      </span>
                    </td>
                  </tr>
                  <tr
                    class="lines-with-lemma__translation"
                  >
                    <td />
                    <td>
                      <span>
                        Ruin su da ho zec la ran hozakan pim lokceg omu dakig sojriuz.
                      </span>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="fade tab-pane"
          role="tabpanel"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <table>
                <tbody>
                  <tr>
                    <th
                      class="lines-with-lemma__genre"
                      colspan="2"
                      scope="col"
                    >
                      Neo-Assyrian
                    </th>
                  </tr>
                  <tr>
                    <th
                      class="lines-with-lemma__header"
                      colspan="2"
                      scope="col"
                    >
                      <span
                        class="lines-with-lemma__textname"
                      >
                        Wama uhro etacafnez ovu wopano vasozli behe halecez pur honuv ziv upepufa rezami ficfiozi veh ti ta.
                      </span>
                       
                      <span>
                        (
                        <a
                          href="https://www.ebl.lmu.de/corpus/Lex/2771/1007298344386560/NA/Pahpa%20muwupdi%20bucelo%20du%20siwele%20mubupel%20januw%20hencan%20iwok%20zo%20fuizi%20pedo%20kemum%20jewkudin%20feksukmab%20bukrepcu."
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          Lex
                           
                          MMDCCLXXI.1007298344386560
                        </a>
                        )
                      </span>
                       
                      <span>
                        Pahpa muwupdi bucelo du siwele mubupel januw hencan iwok zo fuizi pedo kemum jewkudin feksukmab bukrepcu.
                      </span>
                    </th>
                  </tr>
                  <tr
                    class="lines-with-lemma__textline"
                  >
                    <td>
                      <a
                        href="https://www.ebl.lmu.de/corpus/Lex/2771/1007298344386560/NA/Pahpa%20muwupdi%20bucelo%20du%20siwele%20mubupel%20januw%20hencan%20iwok%20zo%20fuizi%20pedo%20kemum%20jewkudin%20feksukmab%20bukrepcu.#1"
                      >
                        1
                      </a>
                    </td>
                    <td>
                      <span
                        class="Transliteration__wordSeparator Transliteration__wordSeparator--AKKADIAN"
                      >
                         
                      </span>
                      <span
                        class="Transliteration__AkkadianWord Transliteration__AkkadianWord--AKKADIAN"
                      >
                        <span
                          class=""
                        >
                          <span
                            class="Transliteration__ValueToken"
                          >
                            <span
                              class=""
                            >
                              kur-kur
                            </span>
                          </span>
                          <sup
                            class="Transliteration__flag"
                          />
                        </span>
                      </span>
                    </td>
                  </tr>
                  <tr
                    class="lines-with-lemma__translation"
                  >
                    <td />
                    <td>
                      <span>
                        Ruin su da ho zec la ran hozakan pim lokceg omu dakig sojriuz.
                      </span>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="fade tab-pane"
          role="tabpanel"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <table>
                <tbody>
                  <tr>
                    <th
                      class="lines-with-lemma__genre"
                      colspan="2"
                      scope="col"
                    >
                      Neo-Assyrian
                    </th>
                  </tr>
                  <tr>
                    <th
                      class="lines-with-lemma__header"
                      colspan="2"
                      scope="col"
                    >
                      <span
                        class="lines-with-lemma__textname"
                      >
                        Wama uhro etacafnez ovu wopano vasozli behe halecez pur honuv ziv upepufa rezami ficfiozi veh ti ta.
                      </span>
                       
                      <span>
                        (
                        <a
                          href="https://www.ebl.lmu.de/corpus/Lex/2771/1007298344386560/NA/Pahpa%20muwupdi%20bucelo%20du%20siwele%20mubupel%20januw%20hencan%20iwok%20zo%20fuizi%20pedo%20kemum%20jewkudin%20feksukmab%20bukrepcu."
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          Lex
                           
                          MMDCCLXXI.1007298344386560
                        </a>
                        )
                      </span>
                       
                      <span>
                        Pahpa muwupdi bucelo du siwele mubupel januw hencan iwok zo fuizi pedo kemum jewkudin feksukmab bukrepcu.
                      </span>
                    </th>
                  </tr>
                  <tr
                    class="lines-with-lemma__textline"
                  >
                    <td>
                      <a
                        href="https://www.ebl.lmu.de/corpus/Lex/2771/1007298344386560/NA/Pahpa%20muwupdi%20bucelo%20du%20siwele%20mubupel%20januw%20hencan%20iwok%20zo%20fuizi%20pedo%20kemum%20jewkudin%20feksukmab%20bukrepcu.#1"
                      >
                        1
                      </a>
                    </td>
                    <td>
                      <span
                        class="Transliteration__wordSeparator Transliteration__wordSeparator--AKKADIAN"
                      >
                         
                      </span>
                      <span
                        class="Transliteration__AkkadianWord Transliteration__AkkadianWord--AKKADIAN"
                      >
                        <span
                          class=""
                        >
                          <span
                            class="Transliteration__ValueToken"
                          >
                            <span
                              class=""
                            >
                              kur-kur
                            </span>
                          </span>
                          <sup
                            class="Transliteration__flag"
                          />
                        </span>
                      </span>
                    </td>
                  </tr>
                  <tr
                    class="lines-with-lemma__translation"
                  >
                    <td />
                    <td>
                      <span>
                        Ruin su da ho zec la ran hozakan pim lokceg omu dakig sojriuz.
                      </span>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="fade tab-pane"
          role="tabpanel"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <table>
                <tbody>
                  <tr>
                    <th
                      class="lines-with-lemma__genre"
                      colspan="2"
                      scope="col"
                    >
                      Neo-Assyrian
                    </th>
                  </tr>
                  <tr>
                    <th
                      class="lines-with-lemma__header"
                      colspan="2"
                      scope="col"
                    >
                      <span
                        class="lines-with-lemma__textname"
                      >
                        Wama uhro etacafnez ovu wopano vasozli behe halecez pur honuv ziv upepufa rezami ficfiozi veh ti ta.
                      </span>
                       
                      <span>
                        (
                        <a
                          href="https://www.ebl.lmu.de/corpus/Lex/2771/1007298344386560/NA/Pahpa%20muwupdi%20bucelo%20du%20siwele%20mubupel%20januw%20hencan%20iwok%20zo%20fuizi%20pedo%20kemum%20jewkudin%20feksukmab%20bukrepcu."
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          Lex
                           
                          MMDCCLXXI.1007298344386560
                        </a>
                        )
                      </span>
                       
                      <span>
                        Pahpa muwupdi bucelo du siwele mubupel januw hencan iwok zo fuizi pedo kemum jewkudin feksukmab bukrepcu.
                      </span>
                    </th>
                  </tr>
                  <tr
                    class="lines-with-lemma__textline"
                  >
                    <td>
                      <a
                        href="https://www.ebl.lmu.de/corpus/Lex/2771/1007298344386560/NA/Pahpa%20muwupdi%20bucelo%20du%20siwele%20mubupel%20januw%20hencan%20iwok%20zo%20fuizi%20pedo%20kemum%20jewkudin%20feksukmab%20bukrepcu.#1"
                      >
                        1
                      </a>
                    </td>
                    <td>
                      <span
                        class="Transliteration__wordSeparator Transliteration__wordSeparator--AKKADIAN"
                      >
                         
                      </span>
                      <span
                        class="Transliteration__AkkadianWord Transliteration__AkkadianWord--AKKADIAN"
                      >
                        <span
                          class=""
                        >
                          <span
                            class="Transliteration__ValueToken"
                          >
                            <span
                              class=""
                            >
                              kur-kur
                            </span>
                          </span>
                          <sup
                            class="Transliteration__flag"
                          />
                        </span>
                      </span>
                    </td>
                  </tr>
                  <tr
                    class="lines-with-lemma__translation"
                  >
                    <td />
                    <td>
                      <span>
                        Ruin su da ho zec la ran hozakan pim lokceg omu dakig sojriuz.
                      </span>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
